### ISSUE TITLE
"Viewport Following" feature in iD

### ISSUE THREAD

**Issue by chtnha** [openstreetmap/iD#3250]
If we want to map very long `highway/river`s  features, we need to pan imagery a lot. It's pain and time consuming. It'd be better if we have tool like [Viewport Following](https://josm.openstreetmap.de/wiki/Help/Action/ViewportFollowing) feature in iD which exists in Java OpenStreetMap Editor (JOSM).

I have shared the illustration here: 

![viewport_following](https://cloud.githubusercontent.com/assets/6770741/16921317/b2109a62-4d2e-11e6-8858-d552822de5bb.gif)

@bhousel @kepta

**Comment by Abbe98** [comment]
I could work on this,  mainly to learn all this new modular iD stuff. If anyone got the time to give me some pointers and be accessible on IRC if I run into something.

I guess we would like a checkbox and if that checkbox is checked we would like to center the map view on every (map)click?

**Comment by bhousel** [comment]
> I guess we would like a checkbox and if that checkbox is checked we would like to center the map view on every (map)click?

Yes, but the checkbox would be a toggle under whatever taskbar comes out of #3123, so it's probably at least a month out.  

If you want to learn modular iD, we could use help with #3180.. Lets start replacing our custom lodash build with only the lodash modules we really need!  (Comment there to coordinate)

**Comment by bhousel** [comment]
Just a quick update that some things in iD do viewport follow now:

* Deleting vertices
* Undo/Redo
* Vertex navigation 
  `{`: first
  `}`: last, 
  `[`: previous, 
  `]`: next, 
  `\`: reparent

Leaving this issue open because we don't yet do viewport following in draw mode.   I have to try it out, and we might want to make it an option.  One concern is what it will do if someone holds down the spacebar and draws a lot of nodes.
